api.py

```diff
@@ -130,13 +113,14 @@ def create_wallet():
 def add_signer():
     wallet_name = request.json['wallet_name']
     signer_name = request.json['signer_name']
-    fingerprint = request.json['device_id']
+    device_id = request.json['device_id']
     wallet = MultisigWallet.open(wallet_name)
-    client, device = get_client_and_device(fingerprint)
-    derivation_path = "m/44h/1h/0h"  # FIXME segwit
-    # FIXME: validate xpub/tpub?
-    xpub = client.get_pubkey_at_path(derivation_path)['xpub']
-    wallet.add_signer(name=signer_name, fingerprint=device['fingerprint'], type=device['type'], xpub=xpub, derivation_path=derivation_path)
+    with get_client_and_device(device_id) as (client, device):
```

Comment:
Switched `get_client_and_device` to a context manager (`with` statement) so that it automatically closes the connection upon exiting the `with` statement. If you leave a connection hanging you won't be able to connect to that device from new client instance -- basically requiring the device to be unplugged and plugged back in